File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {
  int squareOfSum(int n) {
    int sumN = n * (n + 1) ~/ 2;
    return sumN * sumN;
  }
  int sumOfSquares(int n) => n * (n+1) * (2*n + 1) ~/ 6;
  int differenceOfSquares(int n) => squareOfSum(n) - sumOfSquares(n);
}

File: r/difference-of-squares/difference-of-squares.R
square_of_sum <- function(n) { (n * (n + 1) / 2)^2 }

sum_of_squares <- function(n) { n * (n + 1) * (2 * n + 1) / 6 }

difference_of_squares <- function(n) { square_of_sum(n) - sum_of_squares(n) }
